step-plans/step-plans-core/src/main/java/step/core/execution/ExecutionContext.java

```diff
@@ -80,6 +81,7 @@ protected ExecutionContext(String executionId, ExecutionParameters executionPara
 		reportNode.setId(new ObjectId(executionId));
 		reportNodeCache.put(reportNode);
 		setCurrentReportNode(reportNode);
+		put(ExecutionAgentUrls.class, new ExecutionAgentUrls());

```

Comment:
The context map is made for plugins. Don't really see the advantage of not declaring this as a field in ExecutionContext

Reply:
Even better, if that's allowed it will make things even easier.